### ISSUE TITLE
Custom bridge network breaks after reboot

### ISSUE THREAD

**Issue by a-schuurman** [docker/docker#25262]
**Output of `docker version`:**

```
Client:
 Version:      1.11.2
 API version:  1.23
 Go version:   go1.5.4
 Git commit:   b9f10c9
 Built:        Wed Jun  1 21:23:39 2016
 OS/Arch:      linux/amd64

Server:
 Version:      1.11.2
 API version:  1.23
 Go version:   go1.5.4
 Git commit:   b9f10c9
 Built:        Wed Jun  1 21:23:39 2016
 OS/Arch:      linux/amd64
```

**Output of `docker info`:**

```
Containers: 2
 Running: 1
 Paused: 0
 Stopped: 1
Images: 48
Server Version: 1.11.2
Storage Driver: aufs
 Root Dir: /var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 43
 Dirperm1 Supported: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins: 
 Volume: local
 Network: bridge null host
Kernel Version: 3.16.0-4-amd64
Operating System: Debian GNU/Linux 8 (jessie)
OSType: linux
Architecture: x86_64
CPUs: 1
Total Memory: 1.963 GiB
Name: web-dev
ID: WK3O:JC2H:UR2H:ODYW:IM6A:PEMR:Q2AI:YMKB:YYWB:TSVX:ZYMM:UAXW
Docker Root Dir: /var/lib/docker
Debug mode (client): false
Debug mode (server): false
Registry: https://index.docker.io/v1/
WARNING: No memory limit support
WARNING: No swap limit support
WARNING: No kernel memory limit support
WARNING: No oom kill disable support
WARNING: No cpu cfs quota support
WARNING: No cpu cfs period support
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**
Running on VirtualBox since our docker platform is currently in development.

**Steps to reproduce the issue:**
- Add system.d service: /lib/systemd/system/dockerbridge.service:
```bash
[Unit]
Description=Docker Network Service
Before=docker.service
Requires=networking.service

[Service]
Type=notify
ExecStart=/bin/bash -c "if [ $(brctl show | grep dockerbridge0 | wc -l) == 0 ]; then brctl addbr dockerbridge0 && ip addr add 172.18.0.1/16 dev dockerbridge0 && ip link set dev dockerbridge0 up; fi"
TimeoutStartSec=0

[Install]
WantedBy=multi-user.target
```
- Add the service (dockerbridge.service) to 'Requires' in /lib/systemd/system/service.docker. 
- Create the network:
```bash
sudo docker network create \
	--driver=bridge \
	--subnet="172.18.0.0/16" \
	-o "com.docker.network.bridge.default_bridge"="true" \
	-o "com.docker.network.bridge.enable_icc"="true" \
	-o "com.docker.network.bridge.enable_ip_masquerade"="true" \
	-o "com.docker.network.bridge.host_binding_ipv4"="0.0.0.0" \
	-o "com.docker.network.bridge.name"="dockerbridge0" \
	-o "com.docker.network.driver.mtu"="1500" \
	mybridge
```
- Create an image attached to mybridge network:
```bash
sudo docker run --name testcontainer --net mybridge --ip '172.18.0.56' -td imagename
```
Now it actually works and the bridge works fine (so far).
-  **reboot**
-  sudo docker start testcontainer

**Result:**
```bash
$ sudo docker run testcontainer
Error response from daemon: failed to create endpoint testcontainer on network mybridge: network 10813a9f21681bffd0b81bf2b2a69e38ccb5eea1face96fc111978fae0e08b49 does not exist
Error: failed to start containers: testcontainer
```

But the network exists:
```bash
$ sudo docker network ls --no-trunc
NETWORK ID                                                         NAME                DRIVER
bc0a7d77247c42908baa06d22cbe1e9f0241fd32bc1d4b44ffbb4acd8902868a   bridge              bridge              
9d5ef976296b826b33edd054a3367600cc2076263e29c379b535b8650b57e96a   host                host                
3ee5885a05388b6152a60895ad8a3fad89cb97c724c6810780dd01be4b8470c8   none                null                
10813a9f21681bffd0b81bf2b2a69e38ccb5eea1face96fc111978fae0e08b49   mybridge            bridge
```

**Describe the results you received:**
Before the reboot I'm able to run/create a container. But after the reboot it has a similar error upon creating a container attached to this network:

```
$ sudo docker run --name testcontainer --net mybridge --ip '172.18.0.56' -td imagename
docker: Error response from daemon: failed to create endpoint testcontainer on network mybridge: network 10813a9f21681bffd0b81bf2b2a69e38ccb5eea1face96fc111978fae0e08b49 does not exist.
See 'docker run --help'.
```

**Describe the results you expected:**
Docker container running fine since the network exists and the bridge has been created on boot, before docker starts.

**Additional information you deem important (e.g. issue happens only occasionally):**
Always breaks after reboot. But when I remove the network itself, remove the container and add the network again (by using the same `docker network create` command), the container is created and everything works perfectly fine again. It seems very strange to delete and remove a custom bridged network again upon every boot.

**Edit:**
Furthermore quite interesting. Before rebooting the server, the docker container holds the ID of the network and endpoint. But after restarting, it is lost.

*NetworkSettings after creation of the container (docker inspect testcontainer)*:
```json
        "NetworkSettings": {
            "Bridge": "",
            "SandboxID": "47c97d22bd9068ff1f9dcd2a1aa09c848d5268eaadd59fd5fdd92c2ffa628b39",
            "HairpinMode": false,
            "LinkLocalIPv6Address": "",
            "LinkLocalIPv6PrefixLen": 0,
            "Ports": {},
            "SandboxKey": "/var/run/docker/netns/47c97d22bd90",
            "SecondaryIPAddresses": null,
            "SecondaryIPv6Addresses": null,
            "EndpointID": "",
            "Gateway": "",
            "GlobalIPv6Address": "",
            "GlobalIPv6PrefixLen": 0,
            "IPAddress": "",
            "IPPrefixLen": 0,
            "IPv6Gateway": "",
            "MacAddress": "",
            "Networks": {
                "mybridge": {
                    "IPAMConfig": {
                        "IPv4Address": "172.18.0.56"
                    },
                    "Links": null,
                    "Aliases": null,
                    "NetworkID": "10813a9f21681bffd0b81bf2b2a69e38ccb5eea1face96fc111978fae0e08b49",
                    "EndpointID": "b7a5666da9260196f0b9105cf04cfb7672c01abd870bb465dccc9b0894d207c9",
                    "Gateway": "172.18.0.1",
                    "IPAddress": "172.18.0.56",
                    "IPPrefixLen": 16,
                    "IPv6Gateway": "",
                    "GlobalIPv6Address": "",
                    "GlobalIPv6PrefixLen": 0,
                    "MacAddress": "02:42:ac:12:00:38"
                }
            }
        }
```

*NetworkSettings after a reboot*:
```json
        "NetworkSettings": {
            "Bridge": "",
            "SandboxID": "",
            "HairpinMode": false,
            "LinkLocalIPv6Address": "",
            "LinkLocalIPv6PrefixLen": 0,
            "Ports": null,
            "SandboxKey": "",
            "SecondaryIPAddresses": null,
            "SecondaryIPv6Addresses": null,
            "EndpointID": "",
            "Gateway": "",
            "GlobalIPv6Address": "",
            "GlobalIPv6PrefixLen": 0,
            "IPAddress": "",
            "IPPrefixLen": 0,
            "IPv6Gateway": "",
            "MacAddress": "",
            "Networks": {
                "mybridge": {
                    "IPAMConfig": {
                        "IPv4Address": "172.18.0.56"
                    },
                    "Links": null,
                    "Aliases": null,
                    "NetworkID": "",
                    "EndpointID": "",
                    "Gateway": "",
                    "IPAddress": "",
                    "IPPrefixLen": 0,
                    "IPv6Gateway": "",
                    "GlobalIPv6Address": "",
                    "GlobalIPv6PrefixLen": 0,
                    "MacAddress": ""
                }
            }
        }
```

**Comment by a-schuurman** [comment]
So I've just updated to 1.12, this bug seems to be resolved in that version.

**Comment by thaJeztah** [comment]
thanks for the clear bug report, @a-schuurman; I'm closing this issue, as this is resolved in 1.12, but comment if you're able to reproduce this with 1.12
